includes/Core/Conversion_Tracking/Conversion_Tracking.php

```diff
@@ -131,8 +131,25 @@ function ( Conversion_Events_Provider $active_provider ) {
 			}
 		);
 
-		wp_add_inline_script( GTag::HANDLE, 'window._googlesitekit = window._googlesitekit || {};' );
-		wp_add_inline_script( GTag::HANDLE, 'window._googlesitekit.gtagEvent = (name, data) => gtag("event", name, {...data, event_source: "site-kit" });' );
+		$gtag_event = '
+			window._googlesitekit = window._googlesitekit || {};
+			window._googlesitekit.throttledEvents = [];
+			window._googlesitekit.gtagEvent = (name, data) => {
+				var key = JSON.stringify( { name, data } );
+
+				if ( !!window._googlesitekit.throttledEvents[key] ) {
+					return;
+				}
+				window._googlesitekit.throttledEvents[key] = true;
+				setTimeout( () => {
+					delete window._googlesitekit.throttledEvents[key];
+				}, 500 );
```

Comment:
Why such a long timeout here? If a plugin tracks an event multiple times in a short period of time, it seems we'll end up ignoring them. Half a second is brief, but it's totally possible to, for instance, add multiple items to cart in that time by repeatedly clicking on an "Add to cart" button.

Is there any reason we couldn't make this much briefer? Say 10/20/50ms? 🤷🏻‍♂️

Reply:
Thanks, that's a good point, value was in IB, and it slipped my mind that we returned the support for WC, and as you mentioned several events being triggered in shorter time is a realistic possibility. I did several tests with numerous values, and I reached 5ms to be still working. So I updated it for that value.

Reply:
Awesome, thanks! 🙏🏻 

Nice to know we can go that low, given humans can't click that fast I think we're safe 😅